xtask/src/pre-commit

```diff
@@ -1,13 +0,0 @@
-#!/usr/bin/env bash
-
-echo "Running rustfmt and clippy checks."
-
-set -ex
-
-flags=(--all-features --all-targets --workspace)
-cargo fmt -- --check
-if cargo +nightly >/dev/null 2>&1; then
-    cargo +nightly clippy "${flags[@]}" -- --deny warnings
-else
-    cargo clippy "${flags[@]}" -- --deny warnings
-fi
```

Comment:
why does git think this file was removed...

ill fix it

Reply:
...

Reply:
No idea, it has the proper permissions at least so it seems fine.